slowapi/middleware.py

```diff
@@ -0,0 +1,56 @@
+from typing import Union
+
+from starlette.applications import Starlette
+from starlette.middleware.base import (
+    BaseHTTPMiddleware,
+    RequestResponseEndpoint,
+    DispatchFunction,
+)
+from starlette.requests import Request
+from starlette.responses import Response
+from starlette.routing import Route, BaseRoute, WebSocketRoute
+
+from slowapi import Limiter, _rate_limit_exceeded_handler
+
+
+class SlowAPIMiddleware(BaseHTTPMiddleware):
+    async def dispatch(
+        self, request: Request, call_next: RequestResponseEndpoint
+    ) -> Response:
+        app: Starlette = request.app
+        limiter: Limiter = app.state.limiter
+        handler = None
+        for route in app.routes:
+            match, _ = route.matches(request.scope)
+            if match.FULL and hasattr(route, "endpoint"):
+                handler = route.endpoint  # type: ignore
```

Comment:
i couldn't figure out how to make mypy happy here

Reply:
I don't know either. I was hoping that breaking the `if` statement above might help mypy see that your last line is fine, but no. Let's keep it like it is, I think it's ok!